cheng-Cheng/hw_04/0-object-exercise/monkey.js

```diff
@@ -0,0 +1,71 @@
+/* Work with a partner to create objects for three different monkeys (called monkey1, monkey2, monkey3), each with the following properties:
+
+* name
+* species
+* foodsEaten
+
+And the following methods:
+* eatSomething(thingAsString)
+* introduce: producers a string introducing itself, including its name, species, and what it's eaten.
+
+Create 3 monkeys total. Make sure all 3 monkeys have all properties set and methods defined.
+
+Choose which partner will initially take the pair programming role of driver and who will be the observer.
+Switch roles after creating the first object, then again after creating the second.
+Also be sure each partner takes a turn in each role during the exercise steps below.
+
+Exercise your monkeys by retrieving their properties and using their methods.
+Practice using both syntaxes for retrieving properties (dot notation and brackets).
+
+*/
+
+var monkey1 = {
+  name: "Bob",
+  species: "Gorilla",
+  foodsEaten: [],
+  eatSomething: function(food) {
+    monkey1.foodsEaten.push(food)
+    console.log("I have just eaten " + food)
+  },
+  introduce: function() {
+    console.log("Hi my name is " + monkey1.name);
+    console.log("I am a " + monkey1.species);
+    for (i = 0; i < monkey1.foodsEaten.length; i++) {
+    console.log("I have eaten " + monkey1.foodsEaten[i] + " today!");
+    }
+  }
+}
+
+var monkey2 = {
+  name: "Joe",
+  species: "Bonobo",
+  foodsEaten: "Apples",
```

Comment:
Output of monkey2's `foodsEaten`:
```monkey2.introduce()
2016-11-22 20:36:46.029 monkey.js:48 Hi my name is Joe
2016-11-22 20:36:46.029 monkey.js:49 I am a Bonobo
2016-11-22 20:36:46.030 monkey.js:51 I have eaten A today!
2016-11-22 20:36:46.030 monkey.js:51 I have eaten p today!
2016-11-22 20:36:46.030 monkey.js:51 I have eaten p today!
2016-11-22 20:36:46.030 monkey.js:51 I have eaten l today!
2016-11-22 20:36:46.030 monkey.js:51 I have eaten e today!
2016-11-22 20:36:46.030 monkey.js:51 I have eaten s today!```

Monkey2's `foodsEaten` is a string. I'm assuming you meant for it to be an array of strings? What's hilarious is that Monkey2's `eatSomething()` function works because it pushes to Monkey1's array:

```monkey1.introduce()
2016-11-22 20:41:10.022 monkey.js:31 Hi my name is Bob
2016-11-22 20:41:10.022 monkey.js:32 I am a Gorilla
2016-11-22 20:41:10.022 monkey.js:34 I have eaten fruit today!
2016-11-22 20:41:10.022 monkey.js:34 I have eaten apple pie today!
2016-11-22 20:41:10.023 monkey.js:34 I have eaten not a banana lol today! (pushed from Monkey2.eatSomething())```